allSubmissions/AmineSolutions.js

```diff
@@ -0,0 +1,154 @@
+let assert = require('assert');
+console.clear();
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -- - - - - - - - - - - - - - - - - - - //
+
+// Make a function called flipNumbers that takes in an array of strings
+// representing numbers and flips it around. (MEDIUM/HARD)
+
+// Sample Input:
+// ['880', '001', '12345', '0101']
+
+// Sample Output:
+// ['088', '100', '54321', '1010']
+
+// Code here..
+const flipNumbers = (array) => {
+    return array.map(str => {
+      let negative;
+      let decimal;
+      let indexOfDecimal;
+      let outputStr = '';
+  
+      for (let i = str.length - 1; i >= 0; i--) {
+          if (str[i] === '-') {
+            negative = true;
+          }
+          if (str[i] === '.') {
+            decimal = true;
+            indexOfDecimal = i;
+          }
+          if (str[i] !== '-' && str[i] !== '.') {
+            outputStr += str[i];
+          }
+      }
+      if (negative) {
+        outputStr = '-' + outputStr;
+      }
+      if (decimal) {
+        let arr = outputStr.split('');
+        arr.splice(indexOfDecimal, 0, '.');
+        outputStr = arr.join('');
+      }
+      return outputStr;
+    })
+  }
+runFlipMe()
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -- - - - - - - - - - - - - - - - - - - //
+
+// Make a function called magicNumber that takes in an array of numbers and adds the first two,
+// subtracts the third from the total, multiplies the 4th, divides the 5th, etc until you go
+// through the entire array and get a magic number!! (MEDIUM)
+
+// Sample Input:
+// [7,2,3,4,8,6,7]
+
+// Sample Output:
+// 2
+
+// Code here..
+const magicNumber = (array) => {
+    let magicNum = array[0];
+    for (let i = 1; i < array.length; i+= 4) {
+      if (array[i] !== undefined) {
+        magicNum += array[i];
+      } else {
+          break;
+      }
+      if (array[i+1] !== undefined) {
+        magicNum -= array[i+1];
+      } else {
+          break;
+      }
+      if (array[i+2] !== undefined) {
+        magicNum *= array[i+2];
+      } else {
+          break;
+      }
+      if (array[i+3] !== undefined) {
+        magicNum /= array[i+3];
+      } else {
+          break;
+      }
+    }
```

Comment:
It's funny Joey had a similar solution to this but you both made a clean solution good work 👍

Reply:
Thank you :)